### ISSUE TITLE
limit user defined fields

### ISSUE THREAD

**Issue by Parth** [lockbook/lockbook#1228]
malicious people can probably store ungodly amounts of text in strange places like usernames, or filenames. Set some limits in core and server for the overall size of these fields.

**Comment by Parth** [comment]
Identify the places that people get to determine fields that are not documents (usernames and filenames I think are the only things @tvanderstad anything else come to mind?). 

Fields are encrypted so this isn't an exact science, once you've identified a max size hunt down the 2 places (new account, upsert) where these fields are used, introduce a new error for them being too large, and make the code base compile by addressing the errors throughout the stack.

With regards to upsert, upsert is called during a sync, and we don't want a user to successfully create a file locally and then never be able to sync it. So also validate the name size during file creation. 

You're gonna wanna identify a max size unencrypted and a max encrypted size. You can be rough here, but it should never be possible to successfully create a file and then not be able to sync it.

**Comment by Parth** [comment]
(these fields are stored in memory, hence the importance of protecting against abuse)

**Comment by tvanderstad** [comment]
A rogue client could store arbitrary data in anything encrypted, like the user/folder access keys

**Comment by Parth** [comment]
Yeah so you'd probably want to limit those per key to a reasonable amount as well
